packages/flutter/lib/src/painting/fractional_offset.dart

```diff
@@ -588,61 +562,80 @@ class FractionalOffsetDirectional extends FractionalOffsetGeometry {
     }
```

Comment:
We should just move resolve to the base class. (Well, there can be a special version for FractionalOffset that just returns this, but there shouldn't be 3 versions, 2 is enough.)

Reply:
Sure.